### PR TITLE
initial commit for Connect Extenstion

### PR DESCRIPTION
**Pull request by bcleary06**
### 📦 Pull Request

Initial release of Magic Connect Extension

### ✅ Fixed Issues

- n/a

### 🚨 Test instructions

[Sandbox App](https://6t7l7p.csb.app/) for testing

### ⚠️ Don't forget to add a [semver](https://semver.org/) label!

- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @magic-ext/connect@0.1.6-canary.316.3158.0
  # or 
  yarn add @magic-ext/connect@0.1.6-canary.316.3158.0
  ```
</details>

### CODE REVIEW

File: packages/@magic-ext/connect/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# 🔒 Magic Connect Extension for Web Browsers
+
+[![<MagicLabs>](https://circleci.com/gh/magiclabs/magic-js.svg?style=shield)](https://circleci.com/gh/magiclabs/magic-js)
+
+> With the Magic Connect JavaScript SDK extension, you can plug into your favorite social login providers with one, easy-to-use API.
```
Comment (dgerrellsMagic) [inline packages/@magic-ext/connect/README.md:5]: Maybe remove the favorite social login providers bit :D

File: packages/@magic-ext/connect/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "@magic-ext/connect",
+  "version": "2.1.1",
```
Comment (dgerrellsMagic) [inline packages/@magic-ext/connect/package.json:3]: Is this version just copied over from oauth extension? Should we set it as a v1?

Comment (bcleary06) [reply packages/@magic-ext/connect/package.json]: Updated.

File: packages/@magic-ext/connect/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { UserInformation } from './types';
+
+export class ConnectExtension extends Extension.Internal<'connect', any> {
+  name = 'connect' as const;
+  config: any = {
+    mc: true,
+  };
+
+  public showWallet() {
+    const requestPayload = this.utils.createJsonRpcRequestPayload('mc_wallet');
```
Comment (Ethella) [inline packages/@magic-ext/connect/src/index.ts:11]: Can we put payload methods strings to an enum?

Comment (bcleary06) [reply packages/@magic-ext/connect/src/index.ts]: Updated

**Review by hcote — DISMISSED** [review]

**Review by harryEth — DISMISSED** [review]
LGTM just nit

**Review by harryEth — COMMENTED** [review]
nit

File: packages/@magic-ext/connect/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+From your login page:
+
+```ts
+import Web3 from 'web3';
+import { Magic } from 'magic-sdk';
+import { ConnectExtension } from '@magic-ext/connect';
+
+const magic = new Magic('YOUR_API_KEY', {
+  extensions: [new ConnectExtension()],
+  network: testnetOrMainnet
```
[the remark this answers is not in the document]
Comment (bcleary06) [reply packages/@magic-ext/connect/README.md]: Good call. Updated.

**Review by harryEth — APPROVED** [review]

**Comment by magiclabsFE** [comment]
:rocket: PR was released in [`@magic-ext/algorand@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/algorand%404.0.0), [`@magic-ext/avalanche@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/avalanche%404.0.0), [`@magic-ext/bitcoin@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/bitcoin%404.0.0), [`@magic-ext/conflux@2.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/conflux%402.0.0), [`@magic-ext/connect@2.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/connect%402.0.0), [`@magic-ext/cosmos@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/cosmos%404.0.0), [`@magic-ext/flow@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/flow%404.0.0), [`@magic-ext/harmony@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/harmony%404.0.0), [`@magic-ext/icon@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/icon%404.0.0), [`@magic-ext/near@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/near%404.0.0), [`@magic-ext/oauth@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/oauth%403.0.0), [`@magic-ext/polkadot@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/polkadot%404.0.0), [`@magic-ext/react-native-oauth@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-oauth%403.0.0), [`@magic-ext/solana@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/solana%405.0.0), [`@magic-ext/taquito@1.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/taquito%401.0.0), [`@magic-ext/terra@1.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/terra%401.0.0), [`@magic-ext/tezos@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/tezos%404.0.0), [`@magic-ext/webauthn@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/webauthn%403.0.0), [`@magic-ext/zilliqa@4.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/zilliqa%404.0.0), [`@magic-sdk/commons@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%405.0.0), [`@magic-sdk/pnp@3.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/pnp%403.0.0), [`@magic-sdk/provider@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%409.0.0), [`@magic-sdk/react-native@10.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%4010.0.0), [`@magic-sdk/types@8.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%408.0.0), [`magic-sdk@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%409.0.0) :rocket:

**Pull request merged**